machine_learning/linear discriminant analysis.py

```diff
@@ -0,0 +1,389 @@
+        is taken as the prediction.
+
+    Author: @EverLookNeverSee
+
+"""
+
+# importing modules
+from random import gauss
+from math import log
+from os import system, name  # to use < clear > or < cls > commands in terminal or cmd
+
+
+# Making training dataset drawn from a gaussian distribution
+def Normal_gen(mean: float, std_dev: float, instance_count: int) -> list:
+    """ This function generates gaussian distribution instances
+        based-on given mean and standard deviation
+        :param mean: mean value of class
+        :param std_dev: value of standard deviation entered by usr or default value of it
+        :param instance_count: instance number of class
+        :return: a list containing generated values based-on given mean, std_dev and instance_count
+        """
+    generated_instances = []  # An empty list to store generated instances
+    # for loop iterates over instance_count
+    for r in range(instance_count):
+        # appending corresponding gaussian distribution to 'generated_instances' list
+        generated_instances.append(gauss(mean, std_dev))
+
+    return generated_instances
+
+
+# Making corresponding Y flags to detecting classes
+def Y_gen(class_count: int, instance_count: list) -> list:
+    """ This function generates y values for corresponding classes
+    :param class_count: Number of classes(data groupings) in dataset
+    :param instance_count: number of instances in class
+    :return: corresponding values for data groupings in dataset
+    """
+    ys = []  # An empty list to store generated corresponding Ys
+    # for loop iterates over class_count
+    for k in range(class_count):
+        # for loop iterates over related number of instances of each class
+        for p in range(instance_count[k]):
+            # appending corresponding Ys to 'ys' list
+            ys.append(k)
+    return ys
+
+
+# Calculating the class means
+def mean_calc(instance_count: int, items: list) -> float:
+    """ This function calculates given class mean
+    :param instance_count: Number of instances in class
+    :param items: items that related to specific class(data grouping)
+    :return: calculated actual mean of considered class
+    """
+    # the sum of all items divided by number of instances
+    class_mean = sum(items) / instance_count
+    return class_mean
+
+
+# Calculating the class probabilities
+def prob_calc(instance_count: int, total_count: int) -> float:
+    """ This function calculates the probability that a given instance
+        will belong to which class
+        :param instance_count: number of instances in class
+        :param total_count: the number of all instances
+        :return: value of probability for considered class
+        """
+    # number of instances in specific class divided by number of all instances
+    probability = instance_count / total_count
+    return probability
```

Comment:
__def calculate_probabilities(__ or __def probabilities(__ and then just __return instance_count / total_count__